### PR TITLE
Use script from NPM instead of CDN

### LINKED ISSUE #5: Make script loading async?
Currently you're adding the script into the head which will block the page loading until the script is downloaded.

/cc @mickeyreiss-visor

### LINKED ISSUE #6: default ravenjs version is too old and does not stay up to date
Sentry warns that the lib is too old when leaving the default version `3.19.1`:

screen shot 2018-02-18 at 15 00 08

A temporary workaround is to override `version` (currently seems to be `3.22.3`), but it will soon be outdated as well, and updating this can't be automated with tools such as [renovateapp.com](https://renovateapp.com) because the plugin is referring to the CDN, not to the NPM dependency.

The ideal would be for the plugin to use the latest version by default. Could using the NPM dependency solve this?

### PR DESCRIPTION
**Pull request by Thien42**
Solves the issue mentioned in #5. 

Resolves #5. 
Resolves #6.

### CODE REVIEW

**Review by abumalick — COMMENTED** [review]
I don't think it will work as install() will occur in the head before the library is added to the app.

Did you test the PR ?

**Comment by Thien42** [comment]
@abumalick I have updated the PR and we added the plugin to our production setup, to confirm that everything works properly :)

File: src/gatsby-browser.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import Raven    from "raven-js";
+
+exports.onClientEntry = () => {
+    if (window && process.env.NODE_ENV === "production" && window.sentry_config) {
+        window.Raven = Raven;
+
+        window.Raven.config(window.sentry_config.dsn, window.sentry_config.additional_config).install();
```
Comment (abumalick) [inline src/gatsby-browser.js]: Good work 👍 

you can get the plugin options like this too:
```
exports.onClientEntry = (_, pluginParams) => {
  console.log(pluginParams.dsn)
}
```

you can remove all gatsby-ssr file thanks to that.

**Comment by Thien42** [comment]
@abumalick I have updated the PR and removed the `gatsby-ssr.js` file altogether which makes the plugin even simpler indeed :)

**Comment by abumalick** [comment]
A last tip if you want it.

Sorry to give it so late, I just finished implementing it myself.

if you want to load it really async (and separate it from the bundle with code splitting), you can use something like this with gatsby 1.0 : 

```
require.ensure(['raven-js'], require => {
    const Raven = require('raven-js');
    if (!Raven.isSetup()) Raven.config(MY_DSN).install();
   window.Raven = Raven
  });
```

**Review by abumalick — APPROVED** [review]
Good work 👍 

it is very simple now hehe

**Comment by Thien42** [comment]
@abumalick yes indeed ! Not much code left :D

**Review by octalmage — APPROVED** [review]
This is amazing, thanks for working on this!

**Comment by octalmage** [comment]
I'm actually having a hard time getting this to work on my site, I'm getting: 

```
TypeError: Cannot read property 'call' of undefined
```

**Pull request merged**